src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -443,6 +443,10 @@ else if (hasCramInput()) {
                 factory = factory.enable(SamReaderFactory.Option.CACHE_FILE_BASED_INDEXES);
             }
 
+            if ( reads != null ) {
+                reads.close();
+            }
```

Comment:
Yeah, if we keep TwoPassReadWalker that would be the fix, but I'd still like to consider replacing it with `MultiPassReadWalker`. I will spend some time on the latter in the next week or so, under the assumption that this isn't blocking you and is not a high priority. Do let me know if that changes.